tests/exec_name/test

```diff
@@ -51,48 +58,62 @@ ok( $exec  ne "" and $exec2 ne "" );
 );
 
 sub do_test {
-    my $rule = shift(@_);
+    my ( $rules, $filters, $f1, $f2, $f3 ) = @_;
 
     # reset audit
     system("auditctl -D >& /dev/null");
 
     # set the audit-by-executable filter
     my $key = key_gen();
-    system("auditctl -a always,exit -F $rule -F arch=b64 -S exit_group -k $key");
-
-    # run the watched executable
-    system("$exec > /dev/null 2> /dev/null");
+    foreach (@{$rules}) {
+        system("auditctl -a always,exit $_ -F arch=b64 -S exit_group -k $key");
+    }
+    foreach (@{$filters}) {
+        system("auditctl -a exclude,always $_");
+    }
 
-    # run a different executable
+    # run the executables
+    system("$exec1 > /dev/null 2> /dev/null");
     system("$exec2 > /dev/null 2> /dev/null");
+    system("$exec3 > /dev/null 2> /dev/null");
 
     # test if we generate any audit records from the filter rule
     my $result = system("ausearch -i -k $key > $stdout 2> $stderr");
     ok( $result, 0 );
 
     # test if we generate the SYSCALL record correctly
     my $line;
-    my $found_exec = 0;
+    my $found_exec1 = 0;
     my $found_exec2 = 0;
+    my $found_exec3 = 0;
     seek( $fh_out, 0, 0 );
     seek( $fh_err, 0, 0 );
     while ( $line = <$fh_out> ) {
 
         # test if we generate a SYSCALL record
         if ( $line =~ /^type=SYSCALL / ) {
-            if ( $line =~ / exe=$exec / ) {
-                $found_exec = 1;
+            if ( $line =~ / exe=$exec1 / ) {
+                $found_exec1 = 1;
             }
             elsif ( $line =~ / exe=$exec2 / ) {
                 $found_exec2 = 1;
             }
+            elsif ( $line =~ / exe=$exec3 / ) {
+                $found_exec3 = 1;
+            }
         }
     }
-    ok($found_exec and not $found_exec2);
+    ok($found_exec1 == $f1);
+    ok($found_exec2 == $f2);
+    ok($found_exec3 == $f3);
 }
 
-do_test("exe=$exec");
-do_test("exe!=$exec2");
+do_test(["-F  exe=$exec1"],   [], 1, 0, 0);
+do_test(["-F exe!=$exec2"],   [], 1, 0, 1);
+do_test([""], ["-F exe!=$exec1"], 1, 0, 0);
+do_test([""], ["-F  exe=$exec2"], 1, 0, 1);
+do_test(["-F exe=$exec1", "-F exe=$exec2"],   [], 1, 1, 0);
+do_test([""], ["-F exe=$exec1", "-F exe=$exec2"], 0, 0, 1);
```

Comment:
I would like us to try and come up with a better do_test() prototype; the arguments are not easy to quickly read.  At the very least it seems as though do_test() should be able to calculate $f1, $f2, and $f3 based on the desired filter rules, yes?  We could also make better use of Perl's string handling to make things more obvious.  Would something like the following be possible (feel free to come up with your own solution)?

    do_test("always $exec1");
    do_test("always not $exec2");
    do_test("exclude not $exec1");
    do_test("exclude $exec2");
    do_test("always $exec1", "always $exec2");
    do_test("exclude $exec1", "exclude $exec2");

Also, why all the whitespace inconsistency in the arguments?  Is this mandated by `make check-syntax`?